### PR TITLE
Added methods for SpanID and TraceID on bridgeSpanContext

### PR DESCRIPTION
**Pull request by Kaushal28**
Resolves: #3954 

Added methods for SpanID and TraceID on bridgeSpanContext and corresponding unit test case.

### CODE REVIEW

**Review by pellared — COMMENTED** [review]
1. It is not documented how to use it.  See `bridge/opentracing/README.md` and the "extending" section.
2. It would be good to add some tests or at least comments in code that would make sure that someone would not remove these methods in future. 
3. Missing changelog entry.

EDIT: See https://github.com/open-telemetry/opentelemetry-go/pull/3570/ for reference.

**Comment by Kaushal28** [comment]
Updated the PR based on your suggestions. However, I am not exactly sure about the usage of the methods. So I have added a similar description as the reference you provided. So suggestions/feedback would be really helpful there. Also, updated the test case based on that, added comments on the methods and added entry in CHANGELOG.md.

File: bridge/opentracing/bridge_test.go [hunk 1]
```diff
@@ -576,11 +584,8 @@ func TestBridge_SpanContext_TraceID_SpanID(t *testing.T) {
 			spanContext, err := bridge.Extract(ot.TextMap, tmc)
 			assert.NoError(t, err)
 
-			bsc, ok := spanContext.(*bridgeSpanContext)
-			assert.True(t, ok)
-
-			assert.Equal(t, spanID.String(), bsc.SpanID().String())
-			assert.Equal(t, traceID.String(), bsc.TraceID().String())
+			assert.Equal(t, spanID.String(), spanContext.(spanId).SpanID().String())
+			assert.Equal(t, traceID.String(), spanContext.(traceId).TraceID().String())
```
Comment (pellared) [inline bridge/opentracing/bridge_test.go:588]: I propose to put it inside `assert.NotPanics` in case the type assertion panics

**Comment by MadVikingGod** [comment]
I'm wondering, instead of chasing down all the methods that SpanContext exposes, should we just have the bridgeSpanContext be able to produce the otel SpanContext?  

We can't drop IsSampled, but this might head off the need for IsValid, IsRemote, TraceFlags, TraceState, and anything else that might be added in the future.

**Comment by Kaushal28** [comment]
@MadVikingGod, @pellared, I made changes as you suggested. I added `otel.SpanContext` to `bridgeSpanContext` struct as anonymous field and promoted all the available methods on `otel.SpanContext`  to `bridgeSpanContext` so that any method can be accessed similar to how `isSampled()` is currently being accessed. Let me know this change makes sense. Any suggestions/feedback would be really helpful. Thanks.

File: bridge/opentracing/bridge.go [hunk 2]
```diff
@@ -73,17 +73,7 @@ func (c *bridgeSpanContext) ForeachBaggageItem(handler func(k, v string) bool) {
...
-
-// TraceID returns TraceID of underlying span
-func (c *bridgeSpanContext) TraceID() trace.TraceID {
-	return c.otelSpanContext.TraceID()
-}
-
-// SpanID returns SpanID of underlying span
-func (c *bridgeSpanContext) SpanID() trace.SpanID {
-	return c.otelSpanContext.SpanID()
+	return c.SpanContext.IsSampled()
```
Comment (pellared) [inline bridge/opentracing/bridge.go:76]: I think we can remove the definition of `func (c *bridgeSpanContext) IsSampled() bool {`. We get it thanks to type embedding.

File: bridge/opentracing/README.md [hunk 3]
```diff
@@ -58,32 +58,40 @@ if s, ok := sc.(samplable); ok && s.IsSampled() {
...
+- `SpanID() SpanID`
+- `HasSpanID() bool`
+- `WithSpanID(spanID SpanID) SpanContext`
+- `TraceFlags() TraceFlags`
+- `IsSampled() bool`
+- `WithTraceFlags(flags TraceFlags) SpanContext`
+- `TraceState() TraceState`
+- `WithTraceState(state TraceState) SpanContext`
+- `Equal(other SpanContext) bool`
+- `MarshalJSON() ([]byte, error)`
```
Comment (pellared) [inline bridge/opentracing/README.md:97]: I think this is not needed.

File: bridge/opentracing/README.md [hunk 4]
```diff
@@ -58,32 +58,40 @@ if s, ok := sc.(samplable); ok && s.IsSampled() {
 }
 ```
 
-### `SpanContext.TraceID`
-
-Return the underlying OpenTelemetry `Span.TraceID` value by converting a `bridgeSpanContext`.
+Any `SpanContext` method can be accessed as following:
```
Comment (pellared) [inline bridge/opentracing/README.md:61]:
```suggestion
Any [`SpanContext`](https://pkg.go.dev/go.opentelemetry.io/otel/trace#SpanContext) method can be accessed as following:
```

Comment (pellared) [inline bridge/opentracing/README.md:61]: The  "### `SpanContext.IsSampled`" section is now redudant and can be removed. You can just consider adding `IsSampled() bool` to `spanContextProvider` below.

File: bridge/opentracing/bridge.go (hunk 2 above)
Comment (Kaushal28) [reply bridge/opentracing/bridge.go]: Removed.

File: bridge/opentracing/README.md (hunk 3 above)
Comment (Kaushal28) [reply bridge/opentracing/README.md]: Removed.

File: bridge/opentracing/README.md (hunk 4 above)
Comment (Kaushal28) [reply bridge/opentracing/README.md]: Done.

**Comment by pellared** [comment]
```

**Review by pellared — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -20,6 +20,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
   This adds an implementation requirement to set the interface default behavior for unimplemented methods. (#3916)
 - Move No-Op implementation from `go.opentelemetry.io/otel/metric` into its own package `go.opentelemetry.io/otel/metric/noop`. (#3941)
   - `metric.NewNoopMeterProvider` is replaced with `noop.NewMeterProvider`
+- Promoted all the methods from `otel.SpanContext` to `bridgeSpanContext` by embedding `otel.SpanContext` in `bridgeSpanContext`. (#3966)
```
Comment (pellared) [inline CHANGELOG.md:23]:
```suggestion
- Add all the methods from `"go.opentelemetry.io/otel/trace".SpanContext` to `bridgeSpanContext` by embedding `otel.SpanContext` in `bridgeSpanContext`. (#3966)
```

File: bridge/opentracing/README.md [hunk 6]
```diff
@@ -43,17 +43,22 @@ When you have started an OpenTracing Span, make sure the OpenTelemetry knows abo
 
 The bridge functionality can be extended beyond the OpenTracing API.
 
-### `SpanContext.IsSampled`
-
-Return the underlying OpenTelemetry [`Span.IsSampled`](https://pkg.go.dev/go.opentelemetry.io/otel/trace#SpanContext.IsSampled) value by converting a `bridgeSpanContext`.
+Any [`SpanContext`](https://pkg.go.dev/go.opentelemetry.io/otel/trace#SpanContext) method can be accessed as following:
```
Comment (pellared) [inline bridge/opentracing/README.md:46]:
```suggestion
Any [`trace.SpanContext`](https://pkg.go.dev/go.opentelemetry.io/otel/trace#SpanContext) method can be accessed as following:
```

**Review by pellared — APPROVED** [review]

File: bridge/opentracing/bridge_test.go [hunk 7]
```diff
@@ -546,3 +546,49 @@ func TestBridge_SpanContext_IsSampled(t *testing.T) {
 		})
 	}
 }
+
+func TestBridge_SpanContext_Promoted_Methods(t *testing.T) {
```
Comment (MrAlias) [inline bridge/opentracing/bridge_test.go:550]:
```suggestion
func TestBridgeSpanContextPromotedMethods(t *testing.T) {
```

File: bridge/opentracing/bridge_test.go [hunk 8]
```diff
@@ -546,3 +546,49 @@ func TestBridge_SpanContext_IsSampled(t *testing.T) {
...
+	}{
+		{
+			name:    "support for getting spanID and traceID",
+			traceID: [16]byte{byte(1)},
+			spanID:  [8]byte{byte(2)},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
```
Comment (MrAlias) [inline bridge/opentracing/bridge_test.go:576]: This is a lot of boilerplate to run a single test. Instead, can you just run the test directly?

**Comment by Kaushal28** [comment]
@pellared @MrAlias I have fixed the test case as suggested. Please have a look. Thanks.

**Review by pellared — APPROVED** [review]

**Comment by Kaushal28** [comment]
Hi @MrAlias , I have made the required changes, could you please take a look and provide any feedback?

**Comment by MrAlias** [comment]
@Kaushal28 please update this branch with `main` so it can be merged.

**Pull request merged**